### PR TITLE
pre-commit config and lint parse_roms.py

### PR DESCRIPTION
**Pull request by BrianPugh**
This is an initial configuration for using the tool "pre-commit". Basically it runs some static analysis and linting before allowing commits. The gist of it is:

```
pip install pre-commit
pre-commit install
```

If you want to explicitly invoke on everything, run:
```
pre-commit run --all
```

Whenever you make a commit, it will automatically check if the changed files match all the linting rules. If you'd like to skip verification, just do `git commit --no-verify -m "my message here"`.

I only linted the `parse_roms.py` file, but many of the c files need to be linted as well. Would you like me to lint them in this PR?

### CODE REVIEW

**Comment by BrianPugh** [comment]
Things to consider:
* Will the 32-bit addressing patch still work?
* Will tim's `retrogo.patch` still work

File: .flake8 [hunk 1]
```diff
@@ -0,0 +1,5 @@
+[flake8]
+max-line-length = 120
+extend-ignore = E203, W503, E266
```
Comment (kbeckmann) [inline .flake8:3]: I think these ignores can be removed, they seem ok to me.

Comment (BrianPugh) [reply .flake8]: removed

**Comment by kbeckmann** [comment]
Alright. Can you rebase your branch and force push it?

**Comment by kbeckmann** [comment]
Regarding linting C-code, let's not do that now. It will likely make merging external code based on the current code painful.

**Comment by BrianPugh** [comment]
done, and sounds good inregards to the c stuff

**Comment by kbeckmann** [comment]
I split this up and squashed into 9449bdb092dcc01012324fb2d934c664e4cc12f6

**Pull request closed without merging**